osu.Game/Overlays/Settings/Sections/AudioSection.cs

```diff
@@ -22,13 +23,18 @@ public class AudioSection : SettingsSection
 
         public AudioSection()
         {
-            Children = new Drawable[]
+            List<Drawable> children = new List<Drawable>();
+            children.Add(new AudioDevicesSettings());
+            children.Add(new VolumeSettings());
+            children.Add(new OffsetSettings());
+            if (!(RuntimeInformation.IsOSPlatform(OSPlatform.Windows) || RuntimeInformation.IsOSPlatform(OSPlatform.OSX)))
             {
-                new AudioDevicesSettings(),
-                new VolumeSettings(),
-                new OffsetSettings(),
-                new MainMenuSettings()
-            };
+                // Adjusting latency is neither required nor supported on
+                // Windows and Mac.
+                children.Add(new LatencySettings());
+            }
+            children.Add(new MainMenuSettings());
```

Comment:
would probably rewrite the section generation to use `yield return` as it looks pretty bad as done here.

Reply:
Sorry, this is my first time writing any C# code. I looked at https://www.kenneth-truyers.net/2016/05/12/yield-return-in-c/ and didn't see any example that's similar to the code we had before. Could you write out the pattern you had in mind?

Reply:
what I intended to suggest is to have a function that returns the sections in order like so:

```csharp
public IEnumerable<Drawable> getSections()
{
    yield return new AudioDevicesSettings();
    yield return new VolumeSettings();
    yield return new OffsetSettings();

    if (RuntimeInfo.OS == RuntimeInfo.Platform.Linux)
        yield return new LatencySettings();

    yield return new MainMenuSettings();
}
```

and then do `ChildrenEnumerable = getSections()`.

Reply:
Neat. Thanks.